src/components/FormValidator.js

```diff
@@ -41,6 +41,11 @@ export default class FormValidator {
     errorMessageEl.classList.remove(this._errorClass);
   }
 
+  resetValidation() {
+    this._toggleButtonState();
+    this._inputEls.forEach((inputEl) => this._hideInputError(inputEl));
+  }
+
```

Comment:
Please, delete `this._disableSubmitButton();` from line 81 in `FormValidator`


You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.

Reply:
Hi Gennady, I’m still making these changes and will resubmit soon, but came across a couple other things:

In FormValidator.js line 33 config.errorClass = “modal__error_visible”, but this class is not coded in modal.css. Should config.errorClass = “modal__error” not "modal__error_visible"?
![Screen Shot 2024-10-18 at 12 28 55 PM](https://github.com/user-attachments/assets/15c049d9-c82c-4f65-814f-e87fff6673a7)
![Screen Shot 2024-10-18 at 12 29 27 PM](https://github.com/user-attachments/assets/7023265e-4de4-4b48-b803-5446370df728)
![Screen Shot 2024-10-18 at 12 29 56 PM](https://github.com/user-attachments/assets/27967263-aa77-4e3d-9c69-e6daf5dba66b)
![Screen Shot 2024-10-18 at 12 32 25 PM](https://github.com/user-attachments/assets/9e5b5d2f-26af-4f43-8b21-0068d01f3c3e)

Also, on Line 53 should there be an “else” before this._hideInputError(inputEl) ? 
![Screen Shot 2024-10-18 at 12 33 28 PM](https://github.com/user-attachments/assets/95e8befb-3013-4071-ac2a-30e96900926e)

Reply:
> Please, delete `this._disableSubmitButton();` from line 81 in `FormValidator`
> 
> You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.

Should resetValidation() be in this then block?
![Screen Shot 2024-10-18 at 12 38 48 PM](https://github.com/user-attachments/assets/816719ca-01dd-4e02-b584-8fd73b1d424a)

Reply:
> In FormValidator.js line 33 config.errorClass = “modal__error_visible”, but this class is not coded in modal.css. Should config.errorClass = “modal__error” not "modal__error_visible"

If everything works, you don't need to change the code there

Reply:
> Also, on Line 53 should there be an “else” before this._hideInputError(inputEl)

You don't need `else`, if you have `return` (it stops the code)

Reply:
> Should public method resetValidation() be in this then block?

Yes, but it's not a function. It's a method of FormValiator.